### ISSUE TITLE
podman run takes 20 seconds to exit after container exits

### ISSUE THREAD

**Issue by PhracturedBlue** [containers/podman#19915]
### Issue Description

When I run any podman command like: `podman run --rm --network=none -it alpine echo hello` it takes 20 seconds to return a prompt.  The container itself executes and exits immediately as expected, but podman does not return a prompt.  If I press 'Ctrl-C' while waiting, I get a prompt, but while the container is not running, it also hasn't been deleted.  
If I execute a 'podman stop -t -0 ...' on the container, it returns immediately

It doesn't matter if I use the `--rm` or not, the behavior is the same.  Same with the use of `--network=xxx`. Using `podman exec ...` on a running container exits immediately after execution as it should.  `podman stop` and `podman rm` both exit immediately.

I have tried using strace, but strace on the podman process shows nothing interesting, and the conmon process itself exits immediately as expected.

running  `strace -tt podman run --rm alpine echo 1`:
```
13:59:32.571596 rt_sigprocmask(SIG_SETMASK, [KILL STOP], NULL, 8) = 0
13:59:32.571694 futex(0xc00006c948, FUTEX_WAKE_PRIVATE, 1) = 1
13:59:32.571787 futex(0xc00006dd48, FUTEX_WAKE_PRIVATE, 1) = 1
13:59:32.571921 futex(0x5562c722a348, FUTEX_WAIT_PRIVATE, 0, NULL) = ?
13:59:53.008599 +++ exited with 0 +++
```

The 20 seconds is very consistent and seems to be some sort of timeout.

This only happens with one user on the system,  and does not seem to happen when that user runs podman through systemd user-mode.  I am pretty sure that if I reboot, the issue will go away, but I would like to root-cause if possible, as this is a production server, and I would like to not have a re-occurrence.

### Steps to reproduce the issue

Steps to reproduce the issue
```podman run --network=none alpine echo 1```

### Describe the results you received

Podman run hangs for 20 seconds

### Describe the results you expected

`podman run` should return a prompt immediately after container completes

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.31.2
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: cgroupfs
  cgroupVersion: v2
  conmon:
    package: conmon_100:2.1.7-1_amd64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: f633919178f6c8ee4fb41b848a056ec33f8d707d'
  cpuUtilization:
    idlePercent: 98.63
    systemPercent: 0.55
    userPercent: 0.83
  cpus: 4
  databaseBackend: boltdb
  distribution:
    codename: bookworm
    distribution: debian
    version: "12"
eventLogger: file
  freeLocks: 1946
  hostname: nas
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 6.1.46-zfs
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 1800474624
  memTotal: 16267304960
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: podman-aardvark-dns_100:1.7.0-1_amd64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.7.0
    package: podman-netavark_100:1.7.0-1_amd64
    path: /usr/libexec/podman/netavark
    version: netavark 1.7.0
  ociRuntime:
    name: runc
    package: containerd.io_1.6.22-1_amd64                                                                                                          [53/1807]
    path: /usr/bin/runc
    version: |-
      runc version 1.1.8
      commit: v1.1.8-0-g82f18fe
      spec: 1.0.2-dev
      go: go1.19.11
      libseccomp: 2.5.4
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt_0.0~git20230309.7c7625d-1_amd64
    version: |
      pasta unknown version
      Copyright Red Hat
      GNU Affero GPL version 3 or later <https://www.gnu.org/licenses/agpl-3.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_AUDIT_WRITE,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_MKNOD,CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_SETFCAP,CAP_SETGI
D,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns_100:1.2.1-1_amd64
    version: |-
      slirp4netns version 1.2.1
      commit: unknown
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.4
  swapFree: 2708205568
  swapTotal: 3233804288
  uptime: 428h 55m 54.00s (Approximately 17.83 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/ghaushe/.config/containers/storage.conf
  containerStore:
    number: 60
    paused: 0
    running: 13
    stopped: 47
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/ghaushe/.local/share/containers/storage
  graphRootAllocated: 61155995648
  graphRootUsed: 52490702848
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 359
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/ghaushe/.local/share/containers/storage/volumes
version:
  APIVersion: 4.6.1
  Built: 0
  BuiltTime: Wed Dec 31 16:00:00 1969
  GitCommit: ""
  GoVersion: go1.21.0
  Os: linux
  OsArch: linux/amd64
  Version: 4.6.1
```
```


### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

Generic x86_64 PC running Debian bookworm

### Additional information

_No response_

**Comment by Luap99** [comment]
you should run with `strace -f` if you want to see anything useful.

The 20 second timeout is used by podman to wait for the exit code to be stored in the db. This does not seem to happen for some reason. Also adding `--log-level debug` to the podman command should show more info.

**Comment by PhracturedBlue** [comment]
I thought that I had already tried `strace -f` but I was wrong.  However, it didn't provide any additional info besides continually waiting for the FUTEX.

I ran`podman --log-level debug run --rm --network=none alpine echo 1` and see:
```
INFO[0000] Failed to add conmon to cgroupfs sandbox cgroup: creating cgroup path conmon: open /sys/fs/cgroup/cgroup.subtree_control: permission denied
[conmon:d]: failed to write to /proc/self/oom_score_adj: Permission denied

DEBU[0000] Received: 4099378
INFO[0000] Got Conmon PID as 4099367
DEBU[0000] Created container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3 in OCI runtime
DEBU[0000] Attaching to container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3
DEBU[0000] Starting container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3 with command [echo 1]
DEBU[0000] Started container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3
DEBU[0000] Notify sent successfully
1
<...20 second wait here...>
DEBU[0021] Exceeded conmon timeout waiting for container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3 to exit
DEBU[0021] Checking if container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3 should restart
DEBU[0021] Removing container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3
DEBU[0021] Cleaning up container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3
DEBU[0021] Successfully cleaned up container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3
DEBU[0021] Unmounted container "e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3"
DEBU[0021] Removing all exec sessions for container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3
DEBU[0021] Container e3ba44328ba60d5741549dc0daa87818bf30369bff61e48990ab1a3867d9a4f3 storage is already unmounted, skipping...
DEBU[0021] Called run.PersistentPostRunE(podman --log-level debug run --rm --network=none alpine echo 1)
DEBU[0021] Shutting down engines
```
I validated that I get these same 2 failures at the top on a user that does not exhibit this issue.

when comparing `podman info` between a working and non-working user, I discovered that graphRoot was different.  I had moved the rootless containers to a dedicated volume, and set `rootless_storage_path="/mnt/containers/podman/users/$USER"` however this was not taking place for the 'broken' user which was still using storage from ~/.local/containers/storage

Thinking this may be related, I tried a lot of things to convince podman to use the `rootless_storage_path` to no avail.  I ended up rebooting to see if podman would use the proper container path (the answer is still no) but as expected, the reboot did indeed fix the hang condition so I can no longer debug this issue.

I will continue to try to determine why podman ignores the 'rootless_storage_path' for this specific issue (I do not have any overrides in ~/.config/containers) and file a new ticket on that issue if warranted.

**Comment by PhracturedBlue** [comment]
I found that `podman --log-level debug info` showed my graphroot was being overridden, and I found https://github.com/containers/podman/issues/17730 which shows that the bolt_state.db was he culprit.  If I remove the bolt_state.db file, the graphRoot is fixed, but all of my containers and volumes disappear.  It seems this issue is unrelated to the hang, so I'm closing this issue as it is now unreproducible.
